packages/test/test-end-to-end-tests/src/test/offline/stashedOps.spec.ts

```diff
@@ -517,6 +533,116 @@ describeCompat("stashed ops", "NoCompat", (getTestObjectProvider, apis) => {
 		});
 	});
 
+	it("resends insert, delete and move shared array ops", async function () {
+		const pendingOps = await generatePendingState(
+			testContainerConfig,
+			provider,
+			false, // Don't send ops from first container instance before closing
+			async (c, d) => {
+				const array = await d.getSharedObject<ISharedArray<string>>(arrayId);
+				array.insert(0, "test");
+				array.insert(1, "test2");
+				array.insert(2, "test3");
+				array.delete(0);
+				array.insert(0, "test4");
+				array.insert(1, "test5");
+				array.insert(2, "test6");
+				array.insertBulkAfter("test6", ["test7", "test8"]);
+				array.move(0, 3);
+				array.move(1, 4);
+				array.delete(5);
+				array.insert(5, "test9");
+			},
+		);
+
+		// load container with pending ops, which should resend the op not sent by previous container
+		const container2 = await loader.resolve({ url }, pendingOps);
+		const dataStore2 = (await container2.getEntryPoint()) as ITestFluidObject;
+		const array2 = await dataStore2.getSharedObject<ISharedArray<string>>(arrayId);
+		await waitForContainerConnection(container2);
+		await provider.ensureSynchronized();
+		assert.deepEqual(array1.get(), array2.get());
+	});
+
+	it("resends toggle shared array op", async function () {
+		const pendingOps = await generatePendingState(
+			testContainerConfig,
+			provider,
+			false, // Don't send ops from first container instance before closing
+			async (c, d) => {
+				const array = await d.getSharedObject<ISharedArray<string>>(arrayId);
+				let revertible: IRevertible = new SharedArrayRevertible(array, {
+					entryId: "dummy",
+					type: 3,
+					isDeleted: false,
+				} satisfies IToggleOperation);
+				// Attach the revertible event listener.
+				array.on("revertible", (revertibleItem: IRevertible) => {
+					revertible = revertibleItem;
+				});
+				array.insert(0, "test");
+				array.insert(1, "test2");
+				revertible.revert();
```

Comment:
Nit: since it's non-obvious to someone who hasn't looked at SharedArray before, might be nice to add a small inline comment here about the relationship between revert/toggle
```suggestion
				// Reverting the revertible generates a toggle op
				revertible.revert();
```